docs/compilation/performance_tips.md

```diff
@@ -0,0 +1,645 @@
+
+inputset = fhe.inputset(fhe.uint10)
+for lsbs_to_remove in range(0, 10):
+    def f(x):
+        return fhe.round_bit_pattern(x, lsbs_to_remove) // 2
+
+    compiler = fhe.Compiler(f, {"x": "encrypted"})
+    circuit = compiler.compile(inputset)
+
+    print(f"{lsbs_to_remove=} -> {int(circuit.complexity):>13_} complexity")
+```
+
+prints:
+
+```
+lsbs_to_remove=0 -> 9_134_406_574 complexity
+lsbs_to_remove=1 -> 3_209_430_092 complexity
+lsbs_to_remove=2 -> 1_536_476_735 complexity
+lsbs_to_remove=3 -> 1_588_749_586 complexity
+lsbs_to_remove=4 ->   848_133_081 complexity
+lsbs_to_remove=5 ->   525_987_801 complexity
+lsbs_to_remove=6 ->   358_276_023 complexity
+lsbs_to_remove=7 ->   373_311_341 complexity
+lsbs_to_remove=8 ->   400_596_351 complexity
+lsbs_to_remove=9 ->   438_681_996 complexity
+```
+
+### By activating approximate mode for rounding
+
+You can enable [approximate mode](../core-features/rounding.md#exactness) to gain even more performance by sacrificing some more exactness:
+
+```python
+import numpy as np
+from concrete import fhe
+
+inputset = fhe.inputset(fhe.uint10)
+for lsbs_to_remove in range(0, 10):
+    def f(x):
+        return fhe.round_bit_pattern(x, lsbs_to_remove, exactness=fhe.Exactness.APPROXIMATE) // 2
+
+    compiler = fhe.Compiler(f, {"x": "encrypted"})
+    circuit = compiler.compile(inputset)
+
+    print(f"{lsbs_to_remove=} -> {int(circuit.complexity):>13_} complexity")
+
+```
+
+prints:
+
+```
+lsbs_to_remove=0 -> 9_134_406_574 complexity
+lsbs_to_remove=1 -> 5_548_275_712 complexity
+lsbs_to_remove=2 -> 2_430_793_927 complexity
+lsbs_to_remove=3 -> 1_058_638_119 complexity
+lsbs_to_remove=4 ->   409_952_712 complexity
+lsbs_to_remove=5 ->   172_138_947 complexity
+lsbs_to_remove=6 ->    99_198_195 complexity
+lsbs_to_remove=7 ->    71_644_380 complexity
+lsbs_to_remove=8 ->    55_860_516 complexity
+lsbs_to_remove=9 ->    50_978_148 complexity
+```
+
+### By utilizing bit extraction
+
+[Bit extraction](../core-features/bit_extraction.md) is a cheap way to extract certain bits of encrypted values. In can be very useful for improving the performance of circuits.
+
+For example:
+
+```python
+import numpy as np
+from concrete import fhe
+
+inputset = fhe.inputset(fhe.uint6)
+for bit_extraction in [False, True]:
+    def is_even(x):
+        return (
+            x % 2 == 0
+            if not bit_extraction
+            else 1 - fhe.bits(x)[0]
+        )
+
+    compiler = fhe.Compiler(is_even, {"x": "encrypted"})
+    circuit = compiler.compile(inputset)
+
+    if not bit_extraction:
+        print(f"without bit extraction -> {int(circuit.complexity):>11_} complexity")
+    else:
+        print(f"   with bit extraction -> {int(circuit.complexity):>11_} complexity")
+```
+
+prints:
+
+```
+without bit extraction -> 230_210_706 complexity
+   with bit extraction ->  29_506_014 complexity
+```
+
+That's almost 8x improvement to circuit complexity!
+
+## Tweak cryptographic parameters
```

Comment:
"Tweak cryptographic parameters" sounds "you modify it yourself" or "you use something less secure", at least to me. What about changing to "Optimize cryptographic parameters"?

Reply:
Sure, why not.